data-prepper-plugins/mutate-event/src/main/java/com/amazon/dataprepper/plugins/processor/mutateevent/AddEntryProcessor.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.mutateevent;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.processor.AbstractProcessor;
+import com.amazon.dataprepper.model.processor.Processor;
+import com.amazon.dataprepper.model.record.Record;
+
+import java.util.Collection;
+
+@DataPrepperPlugin(name = "add_entry", pluginType = Processor.class, pluginConfigurationType = AddEntryProcessorConfig.class)
+public class AddEntryProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    private final AddEntryProcessorConfig config;
+
+    @DataPrepperPluginConstructor
+    public AddEntryProcessor(final PluginMetrics pluginMetrics, final AddEntryProcessorConfig config) {
+        super(pluginMetrics);
+        this.config = config;
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        for(final Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+            final String key = config.getKey();
+            final Object value = config.getValue();
```

Comment:
We don't need to fetch and assign these values for every event (or for every doExecute call). We can push this to the constructor.

```
    private final boolean overwriteIfKeyExists;

    private final String key;

    private final Object value;

    @DataPrepperPluginConstructor
    public AddEntryProcessor(final PluginMetrics pluginMetrics, final AddEntryProcessorConfig config) {
        super(pluginMetrics);
        this.overwriteIfKeyExists = config.getOverwriteIfKeyExists();
        this.key = config.getKey();
        this.value = config.getValue();
    }

    @Override
    public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
        for(final Record<Event> record : records) {
            final Event recordEvent = record.getData();
            if(!recordEvent.containsKey(key) || overwriteIfKeyExists) {
                recordEvent.put(key, value);
            }
        return records;
    }
```

The same approach can be applied to the other processors as well.